src/transports/encodePolyfill.ts

```diff
@@ -1,14 +1,21 @@
+import { SDK_VERSION } from '@sentry/core';
+
 import { CAP_GLOBAL_OBJ } from '../utils/webViewUrl';
 import { utf8ToBytes } from '../vendor';
 
 export const useEncodePolyfill = (): void => {
-  if (!CAP_GLOBAL_OBJ.__SENTRY__) {
-    (CAP_GLOBAL_OBJ.__SENTRY__ as Partial<
+  const globalCarriers = CAP_GLOBAL_OBJ.__SENTRY__;
+
+  if (!globalCarriers) {
+    (globalCarriers as Partial<
       (typeof CAP_GLOBAL_OBJ)['__SENTRY__']
     >) = {};
   }
+  const capacitorSiblingCarrier = CAP_GLOBAL_OBJ.__SENTRY__?.[SDK_VERSION];
 
-  CAP_GLOBAL_OBJ.__SENTRY__.encodePolyfill = encodePolyfill;
+  if (capacitorSiblingCarrier) {
+    capacitorSiblingCarrier.encodePolyfill = encodePolyfill;
+  }
```

Comment:
Because this is called before the browserInit the `capacitorSiblingCarrier ` won't be set and thus the `encodePolyfill` won't be set.

JS has a helper function https://github.com/getsentry/sentry-javascript/blob/fc0af3892a2704326d2338c05d3f049176d9358d/packages/core/src/carrier.ts#L42 which ensures the version carrier is always ready.

If it's not ready we should log an error because that's unexpected.

Reply:
Sure thing I will use it, although I am not sure if it will always be ready according to the function comment  `FIXME: This function is problematic`

Reply:
The reason for the `FIXME: This function is problematic` is that the type still returns the carrier marked as optional, even tho it's created when missing.

But it's used in the JS SDK, so I think we should use it too or ensure in a different way that the carrier is created.

Reply:
I see, using main carrier on https://github.com/getsentry/sentry-capacitor/pull/840/commits/2e8c915868248d9d56426313d69a7a7a0368548d